File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/ConfigurationMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.model.RookConfiguration
import com.rookmotion.rook_sdk_health_connect.proto.ConfigurationProto

fun ConfigurationProto.toRookConfiguration(): RookConfiguration {
    return RookConfiguration(
        clientUUID,
        secretKey,
        environment.toRookEnvironment(),
    )
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/configuration_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/mapper/environment_mappers.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pb.dart';
import 'package:rook_sdk_health_connect/src/domain/util/check_non_default.dart';

extension RookConfigurationMappers on RookConfiguration {
  ConfigurationProto toProto() {
    return ConfigurationProto(
      clientUUID: clientUUID,
      secret: secret,
      environment: environment.toProto(),
      enableBackgroundSync: enableBackgroundSync,
      appId: appId ?? defaultString,
    );
  }
}
